app/src/main/java/app/revanced/integrations/youtube/patches/components/DescriptionComponentsFilter.java

```diff
@@ -45,21 +44,44 @@ public DescriptionComponentsFilter() {
                 "transcript_section"
         );
 
+        macroMarkersCarousel = new StringFilterGroup(
+                null,
+                "macro_markers_carousel.eml"
+        );
+
+        chaptersSectionBufferGroup = new ByteArrayFilterGroup(
+                Settings.HIDE_CHAPTERS_SECTION,
+                "chapters_horizontal_shelf"
+        );
+
+        keyConceptsSectionBufferGroup = new ByteArrayFilterGroup(
+                Settings.HIDE_KEY_CONCEPTS_SECTION,
+                "learning_concept_macro_markers_carousel_shelf"
+        );
+
         addPathCallbacks(
                 attributesSection,
-                chaptersSection,
                 infoCardsSection,
                 podcastSection,
-                transcriptSection
+                transcriptSection,
+                macroMarkersCarousel
         );
     }
 
-
     @Override
     boolean isFiltered(@Nullable String identifier, String path, byte[] protobufBufferArray,
                        StringFilterGroup matchedGroup, FilterContentType contentType, int contentIndex) {
         if (exceptions.matches(path)) return false;
+        if (matchedGroup == macroMarkersCarousel) {
+            if (contentIndex == 0
+                    && chaptersSectionBufferGroup.check(protobufBufferArray).isFiltered()
+                    && keyConceptsSectionBufferGroup.check(protobufBufferArray).isFiltered()) {
```

Comment:
Do both options have to be enabled?

Reply:
Not sure, does only having one enabled be enough to hide both?

Reply:
I don't know what you mean by both, but I do not see why `Settings.HIDE_CHAPTERS_SECTION` should control hiding "Key concepts"

Reply:
For hiding key concepts, there is `Settings.HIDE_KEY_CONCEPS_SECTION` as well. Both buffergroup needs to be defined right?

Reply:
It's present in the issue [#3324](https://github.com/ReVanced/revanced-patches/issues/3324)